Implement performance metrics BehaveTree Node

goudcode opened the pull request:
## Description

I've implemented the performance per tree in the Godot engine profiler.
Time is being showed in nanosecconds and tree will be automatically added and removed to the profiler.

It might be better to disable all the metric code when running in release mode, example: use `if OS.is_debug_build()`
Any ideas are welcome.

## Addressed issues
#89 

## Screenshots
![afbeelding](https://user-images.githubusercontent.com/23435053/213942849-0aac2de6-a19d-412a-891d-52e82ea9634c.png)

goudcode committed:
Implement performance metrics BehaveTree Node

bitbrain commented on addons/beehave/nodes/beehave_tree.gd:
```diff
@@ -22,6 +22,8 @@ enum {
 
 var actor : Node
 
+var _tree_metric_name : String
```
we may want to give this a better name: as it is tracking the `process_time` of this tree, perhaps something like this:
```suggestion
var _process_time_metric_name : String
```
(we are inside the tree already so using "tree" is redundant)

bitbrain commented on addons/beehave/nodes/beehave_tree.gd:
```diff
@@ -93,3 +100,22 @@ func enable() -> void:
+func _enter_tree() -> void:
+	# Get the name of the parent node name for metric
+	var parent_name = get_parent().name
+	_tree_metric_name = "beehave/process_time/%s(%s)" % [parent_name, get_instance_id()]
```
From a structural perspective, we may want to have multiple metrics as part of the same tree. Therefore, I'd group them differently:
```suggestion
	_tree_metric_name = "beehave/%s-%s/process_time" % [parent_name, get_instance_id()]
```

bitbrain commented on addons/beehave/nodes/beehave_tree.gd:
```diff
@@ -93,3 +100,22 @@ func enable() -> void:
+
+func _enter_tree() -> void:
+	# Get the name of the parent node name for metric
+	var parent_name = get_parent().name
```
Using `get_parent()` here is wrong, as there are different ways to associate a node to a tree (see `actor_node_path` export property)
```suggestion
	var parent_name = actor.name
```

bitbrain reviewed (changes requested):
Very good start! Apart from a few smaller comments this looks generally good.

Could we also add additional metrics like described on #89?

| Metric Name | Description |
|---|---|
| `beehave/node_count` | The number of total beehave nodes in the scene |
| `beehave/active_count` | Number of active behaviour trees in the scene |

Also reminder, that there is no need to check for debug mode, as Godot will only query the monitors inside the editor anyways and ensures not to call these in production mode.

bitbrain commented on addons/beehave/nodes/beehave_tree.gd:
```diff
@@ -93,3 +100,22 @@ func enable() -> void:
+
+
+# Called by the engine to profile this tree
+func _get_tree_metric() -> float:
```
Oh, this method also should be renamed to something like this:
```suggestion
func _get_process_time_metric_value() -> float:
```

goudcode committed:
Refactored metric variable names

- Refactored variable names to tell what the metric is.
- Parent name is now obtained by actor property

goudcode commented:
The metric is now registered in the ready() function instead of the _enter_tree().
The reason for this is that the actor was not known in that function.

bitbrain reviewed (approved):
Thank you for your contribution!
